### PR TITLE
fix: ensure Kubernetes/Docker container info is captured

### LINKED ISSUE #950: Kubernetes metadata isn't picked up automatically

### LINKED ISSUE #996: An in-range update of elastic-apm-http-client is breaking the build 🚨
## The dependency [elastic-apm-http-client](https://github.com/elastic/apm-nodejs-http-client) was updated from `7.2.0` to `7.2.1`.

🚨 [View failing branch](https://github.com/elastic/apm-agent-nodejs/compare/master...elastic:greenkeeper%2Felastic-apm-http-client-7.2.1).

This version is **covered** by your **current version range** and after updating it in your project **the build failed**.

elastic-apm-http-client is a direct dependency of this project, and **it is very likely causing it to break**. If other packages depend on yours, this update is probably also breaking those in turn.

<details>
<summary>Status Details</summary>

- ❌ **continuous-integration/travis-ci/push:** The Travis CI build is in progress ([Details](https://travis-ci.org/elastic/apm-agent-nodejs/builds/518224745?utm_source=github_status&utm_medium=notification)).
- ✅ **continuous-integration/appveyor/branch:** AppVeyor build succeeded ([Details](https://ci.appveyor.com/project/watson/apm-agent-nodejs/builds/23737437)).
- ❌ **continuous-integration/jenkins/branch:** This commit has test failures ([Details](https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/greenkeeper%252Felastic-apm-http-client-7.2.1/1/display/redirect)).
</details>

---

<details>
<summary>Commits</summary>
<p>The new version differs by 2 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/elastic/apm-nodejs-http-client/commit/2b5c6b756c6879b29ad3cdfb85096b9c80df535a"><code>2b5c6b7</code></a> <code>7.2.1</code></li>
<li><a href="https://urls.greenkeeper.io/elastic/apm-nodejs-http-client/commit/2e07a5df7bbe268284a8d0ed557f1d293096f6fa"><code>2e07a5d</code></a> <code>fix(package): bump container-info to v1.0.1 (#54)</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/elastic/apm-nodejs-http-client/compare/239e39a3ad30cbd8c3105d3873b7fee85d0e6f88...2b5c6b756c6879b29ad3cdfb85096b9c80df535a">full diff</a></p>
</details>

<details>
<summary>FAQ and help</summary>

There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---

Your [Greenkeeper](https://greenkeeper.io) Bot :palm_tree:

### PR DESCRIPTION
**Pull request by watson**
Fixes #950
Closes #996

### CODE REVIEW

**Comment by watson** [comment]
Once the tests clear, I'll merge this without a review as the tests in master are currently broken because of this

File: test/agent.js [hunk 1]
```diff
@@ -1176,11 +1178,25 @@ function assertMetadata (t, payload) {
...
+  t.equal(payload.system.hostname, os.hostname())
+  t.equal(payload.system.architecture, process.arch)
+  t.equal(payload.system.platform, process.platform)
+
+  if (inContainer) {
+    t.deepEqual(Object.keys(payload.system.container), ['id'])
+    t.equal(typeof payload.system.container.id, 'string')
+    t.ok(/^[\da-f]{64}$/.test(payload.system.container.id))
+  } else {
+    // dummy asserts to ensure that the t.plan is as expected
```
Comment (Qard) [inline test/agent.js]: Maybe just change the `asserts` count on this function when in a container?

Comment (watson) [reply test/agent.js]: Doh, you're right. I was thinking it could change between tests, but obviously it can't

Comment (watson) [reply test/agent.js]: Done: #998

**Pull request merged**